src/redux/preview.js

```diff
@@ -556,3 +561,28 @@ module.exports.updateProject = (id, jsonData, username, token) => (dispatch => {
         dispatch(module.exports.setProjectInfo(body));
     });
 });
+
+module.exports.reportProject = (id, jsonData, username, token) => (dispatch => {
+    dispatch(module.exports.setFetchStatus('report', module.exports.Status.FETCHING));
+    // scratchr2 will fail if no thumbnail base64 string provided. We don't yet have
+    // a way to get the actual project thumbnail in www/gui, so for now just submit
+    // a minimal base64 png string.
+    defaults(jsonData, {
+        thumbnail: 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC' +
+            '0lEQVR42mP8/x8AAwMCAO+ip1sAAAAASUVORK5CYII='
+    });
+    api({
+        host: '',
+        uri: `/site-api/projects/all/${id}/report/`,
+        authentication: token,
+        method: 'POST',
+        json: jsonData,
+        useCsrf: true
+    }, (err, body, res) => {
+        if (err || res.statusCode >= 300) {
```

Comment:
What is the `3XX` case?

Reply:
In development, we were seeing that in some cases scratchr2 may return a 3XX, while not adding the report to the db. If we see that, we should consider the report submission to have failed.

Reply:
You could make a case that this should be >= 204 or something else...

Reply:
I think the case I would make is that `res.statusCode !== 200` is the error state. I think this is fine as-is though.